app/controllers/pages_controller.rb

```diff
@@ -4,6 +4,6 @@ class PagesController < ApplicationController
   skip_before_action :disable_caching
 
   def page_not_found
-    render layout: false, status: 404
+    render layout: false, status: 404, formats: %i[css html png]
```

Comment:
I don't know how, but Rails handles those automatically without having to specify them. In fact, after testing this again, it looks like all we need is to specify the `html` format, and it will generate a 404 for any extension.